src/utils/withUserCheck.ts

```diff
@@ -0,0 +1,80 @@
+import { User } from '@supabase/supabase-js'
+import { GetServerSidePropsContext } from 'next'
+
+import { supabase } from '../lib/supabase'
+
+type CheckOptions = {
+  authenticated?: boolean
+  onboarded?: boolean
+  redirectTo?: string
+}
+
+enum UserStatus {
+  None,
+  Authenticated,
+  Onboarded
+}
+
+const getUserStatus = async (user?: User | null): Promise<UserStatus> => {
+  const email = user?.email
+
+  if (!email) {
+    return UserStatus.None
+  }
+
+  const { data } = await supabase.from('User').select('*').eq('email', email)
```

Comment:
```suggestion
  const { data } = await supabase.from('User').select('*').eq('email', email).single()
```

This will fetch just one record.